libcylinder/src/error.rs

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
There's a lot of unrelated things happening in this commit. Error types created, functions moved that are unrelated to the primary purpose of the commit.

Reply:
Split the commit in two: one for moving the hex functions and creating its error type, one for adding the signer/verifier interface and their error types.